frontend/src/components/auth/Login.tsx

```diff
@@ -38,25 +42,42 @@ const Login = (): React.ReactElement => {
   }
 
   const onLogInClick = async () => {
-    const user: AuthenticatedUser = await authAPIClient.login(email, password);
-    const isUserVerified = user?.accessToken
-      ? await authAPIClient.isUserVerified(email, user.accessToken)
-      : null;
-    if (!user || !isUserVerified) {
-      // will need to change this for different errors
-      // eslint-disable-next-line no-alert
-      alert("Bad login, user not found");
-      return;
-    }
-    if (user.role.toLowerCase() !== role.toLowerCase()) {
-      // change this later to not use an alert
-      // eslint-disable-next-line no-alert
-      alert(`Bad login. Expected ${user.role}, got ${role}`);
-      return;
-    }
+    try {
+      setLoginError(undefined);
+      setEmailError(undefined);
+      const user: AuthenticatedUser = await authAPIClient.login(
+        email,
+        password,
+      );
+      const isUserVerified = user?.accessToken
+        ? await authAPIClient.isUserVerified(email, user.accessToken)
+        : null;
+      if (!user || !isUserVerified) {
+        // will need to change this for different errors
+        // eslint-disable-next-line no-alert
+        alert("Bad login, user not found");
+        return;
+      }
+      if (user.role.toLowerCase() !== role.toLowerCase()) {
+        // change this later to not use an alert
+        // eslint-disable-next-line no-alert
+        alert(`Bad login. Expected ${user.role}, got ${role}`);
+        return;
+      }
 
-    localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
-    setAuthenticatedUser(user);
+      localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
+      setAuthenticatedUser(user);
+    } catch (e: unknown) {
```

Comment:
Propagating the error by continuously throwing it and then catching it in the frontend works well here.

However, it may also be nice to have some default response type for our backend calls. IE
```
type Response = {
data: any
error: any
}
```

Rationale: For more complex processes, we might not want to throw errors and then have a general catch case as above. If we are doing a database update with multiple steps, if one function fails we might need to perform some recovery action.

```
def updateDB():
   const {data: task1Data, error: task1Error} = performTask1()
   if (task1Error) {
       emergencyRecoveryTask()
   }
   else {
        const {data: task2Data, error: task2Error} = performTask2(task1Data)
        ...
   }
   ```
   
 SAMPLE Backend Function
 
 ```
 performTask1():
      try {
         // something
         const coolData = ....
         return { data: coolData, error: None }
       } catch (e) {
         return {data: None, error: e }
       }

Reply:
Yea that works ! I see now, you can definitely make it work either way.